src/persistence.ts

```diff
@@ -0,0 +1,59 @@
+import { ViolationInfo } from "./lint";
+import { CommitUser } from "types/github";
+import { createCommitDiagnosis } from "./db/models/CommitDiagnosis";
+import { createCommit, findCommit } from "./db/models/Commit";
+import {
+  findUserByEmail,
+  findUserByName,
+  createUser,
+  User,
+} from "./db/models/User";
+
+export interface CommitInfo {
+  author: CommitUser;
+  commit: {
+    id: string;
+    message: string;
+    score: number;
+    timestamp: string;
+    violations: ViolationInfo[];
+  };
+}
+
+async function findOrCreateUser(user: CommitUser): Promise<User> {
+  const existingUser =
+    (await findUserByName(user.name)) ?? (await findUserByEmail(user.email));
+  if (!existingUser) {
+    return createUser({
+      email: user.email,
+      name: user.name,
+    });
+  }
+  return existingUser;
+}
+
+export async function saveCommit(commitInfo: CommitInfo): Promise<void> {
```

Comment:
This file can be moved into `db` and make it `db/index.ts`, since this is the main interface that other modules would use. The database can be exported in `db/db.ts` instead.